### DISCUSSION TITLE
schema rule unhappy about become_method from another collection

### DISCUSSION THREAD

**Discussion by Dunedan** [ansible/ansible-lint#3170] [Q&A]
I have the feeling that I'm missing something obvious, but I can't get ansible-lint to accept a `become_method` from another collection for the `schema[tasks]` rule.

Here is a minimal example to reproduce this behavior:

```yaml
- name: Show failing schema[tasks] rule
  ansible.builtin.debug:
    msg: "hello world"
  become: true
  become_method: community.general.sudosu
  become_user: someuser
```

When encountering this task, ansible-lint exits with the following error message:
```
schema[tasks]: {'name': 'Show failing schema[tasks] rule', 'ansible.builtin.debug': {'msg': 'hello world'}, 'become': True, 'become_method': 'community.general.sudosu', 'become_user': 'someuser'} is not valid under any of the given schemas
roles/test/tasks/main.yml:1  Returned errors will not include exact line numbers, but they will mention
the schema name being used as a tag, like ``schema[playbook]``,
``schema[tasks]``.

This rule is not skippable and stops further processing of the file.

If incorrect schema was picked, you might want to either:

* move the file to standard location, so its file is detected correctly.
* use ``kinds:`` option in linter config to help it pick correct file type.


Read documentation for instructions on how to ignore specific rule violations.

              Rule Violation Summary              
 count tag           profile rule associated tags 
     1 schema[tasks] basic   core                 

Failed after min profile: 1 failure(s), 0 warning(s) on 3 files
```

Looking at the [JSON Schema for `become_method`](https://github.com/ansible/ansible-lint/blob/27e4ae138521ab3f87e189d4b405b3f7f309e231/src/ansiblelint/schemas/tasks.json#L4-L26), there doesn't seem to be a way to make ansible-lint happy, but there surely must be one.

What's the recommended solution for this situation?

**Comment by dlefevre** [comment]
I'm curious about this as well. I'm using `become_method: containers.podman.podman_unshare` and there's no way to get it validated against the tasks schema. 

The only workaround I've found so far is to exclude the path of the offending yaml in `.ansible-lint`. However, that's kinda throwing out the baby with the bathwater.

**Comment by guppy0130** [comment]
Looks like any changes would go here (assuming the comment about generation from `ansible.json` still holds?): https://github.com/ansible/ansible-lint/blob/42329279496085a30946c1c93fc2fa8e0dbb2d5b/src/ansiblelint/schemas/ansible.json#L43-L65

I see `become_method` takes jinja though:

https://github.com/ansible/ansible-lint/blob/42329279496085a30946c1c93fc2fa8e0dbb2d5b/src/ansiblelint/schemas/ansible.json#L60-L62

so something silly like

```yaml
  become_method: "{{ 'ansible.netcommon.enable' }}"
```

will pass, but that's not great
